src/sspak-sniffer.php

```diff
@@ -29,10 +29,17 @@
 } else if(file_exists(BASE_PATH.'/framework/core/Core.php')) {
 	//SS 3.x
 	require_once(BASE_PATH. '/framework/core/Core.php');
-} else if(file_exists(BASE_PATH.'/framework/src/Core/Core.php')) {
+} else if(file_exists(BASE_PATH.'/vendor/silverstripe/framework')) {
 	//SS 4.x
 	require_once(BASE_PATH. '/vendor/autoload.php');
-	require_once(BASE_PATH. '/framework/src/Core/Core.php');
+	$kernel = new SilverStripe\Core\CoreKernel(BASE_PATH);
+	//boot the parts of the kernel to populate the DB config
+	foreach (array('bootDatabaseEnvVars', 'bootDatabaseGlobals') as $bootMethod) {
```

Comment:
We could just do ->boot() instead of extracting the protected methods. It may be less risky as it doesn't bypass internal business logic.

For instance, your solution would omit all `_config.php` modifications to db details.

Reply:
@phptek 's solution uses a whole `Application` object, which is fine but probably not necessary. :)

https://github.com/silverstripe/sspak/compare/master...phptek:master#diff-741063a460d037197c73e0c8347c5618R58

Reply:
I wanted to stop it from building an entire manifest cache just to extract DB details.

I'm deliberately aiming for the minimum needed to extract the info we need